tasks/main.yaml

```diff
@@ -11,6 +11,15 @@
       set_fact:
         rsyslog_purge_original_conf: "{{ logging_purge_confs|d(false) }}"
 
+    - name: Set use_python3 to false if python3 is not installed.
+      set_fact:
+        use_python3: false
+      when: "ansible_python_version is version('3', '<')"
+
+    - name: Check python version
+      debug:
+        msg: "Python version is {{ ansible_python_version }}; use_python3 is {{ use_python3 }}"
+
```

Comment:
I don't think that this is something specific to the logging role, so if it is really needed, it should be solved for all the roles in a common way.

Reply:
+1
But do you think my expectation is valid?  That is, we will see ansible_python_version 3.X here if we install python3 on centos 7 with molecule Dockerfile?
If it is, shall I open an issue on JIRA?

Reply:
First of all, why do you worry about the python version? What is the problem you are trying to solve?

Reply:
Ok.  What I need to make sure is the target host has the python libselinux package.  When I was testing the first cut of LSR/Logging, the package was not in the target host and my attempt to deploy config failed due to the error [2].  So, I included the package name in rsyslog_base_packages (see [1]).  Unfortunately, the package name was changed from libselinux-python to python3-libselinux starting from RHEL8 [3].  So, I had to add this trick [1] to get the right libselinux package name.  That's the reason why I was worried about the python version.

Regarding the python3 on centos 7, I'm updating molecule Dockerfile.j2 like this [4].  I thought python3 is used on all testing platforms including centos 7, but the ansible log says python2 is being used, which made me confused and asked the question.

[1] https://github.com/linux-system-roles/logging/pull/80/files#diff-def9bf9161e11520d46eea7aa85093e1R59
[2] "msg": "Aborting, target uses selinux but python bindings (libselinux-python) aren't installed!"
[3] https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/considerations_in_adopting_rhel_8/changes-to-packages_considerations-in-adopting-rhel-8
[4] https://github.com/linux-system-roles/logging/pull/80/files#diff-a8de8e86be756e09de886c06dbf13c06

Reply:
Investigated some more on this python selinux library issue and figured out the logging_debug added in this commit required libselinux-python for rhel7 and python3-libselinux for rhel8.
https://github.com/linux-system-roles/logging/commit/3bc23d034a8deee0ab34eeaae64a6f58cb62e44e

We may be able to avoid the selinux issue if we choose the path carefully...

Reply:
@nhosoi I am curious how to reproduce the problem, preferably with a test.

Reply:
@pcahyna 
> @nhosoi I am curious how to reproduce the problem, preferably with a test.

A good question. :)  On my laptop, I happen to have these images [1] and each has libselinux-python-2.5-14.1.el7.x86_64 and python3-libselinux-2.9-2.1.el8.x86_64 preinstalled, respectively.  That means we don't need to install the python libselinux by ourselves.  When we wrote the Logging role, our target host was RHCOS, which is lean enough not to include the python selinux library.  I think the issue is how much we could rely on the preinstalled packages on the target host.  @sradco, in your use case, do you have a chance to run Logging role against a target host which does not have python selinux library?
```
[1]
CentOS-7-x86_64-GenericCloud.qcow2c
CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2
```

Reply:
Is RHCOS still a target that we need to support? We never had such a request for the other system roles, is logging special in this respect? Does RHCOS even have Python installed?

Reply:
> Is RHCOS still a target that we need to support? We never had such a request for the other system roles, is logging special in this respect? Does RHCOS even have Python installed?

I think we should not support RHCOS, but let's wait and see if others want to comment.

Reply:
@sradco do you recall why there was a need to support RHCOS?